packages/flutter_tools/lib/src/platform_plugins.dart

```diff
@@ -41,9 +44,13 @@ abstract class VariantPlatformPlugin {
   Set<PluginPlatformVariant> get supportedVariants;
 }
 
-abstract class NativeOrDartPlugin {
+abstract class NativeOrDartOrFfiPlugin {
+  /// Determines whether the plugin has a FFI implementation or if it's a
+  /// Dart-only or native plugin.
+  bool isFfi();
+
   /// Determines whether the plugin has a native implementation or if it's a
-  /// Dart-only plugin.
+  /// Dart-only or FFI plugin.
```

Comment:
`isNative` is used to determine whether plugin registry needs to be generated.

I completely agree that we should hammer out terminology.
E.g. is an FFI plugin a plugin? If the answer is yes, how do we call non-FFI plugins? Dart source code plugins are also called plugins.
If we find a nice naming, I can do the refactoring to make everything consistent again in the code base.

For example:
* plugins
  * dart plugins
  * native plugins
    * non-ffi plugins (bad name imo)
    * ffi plugins

Reply:
Or should we not call it a plugin at all?

* packages
  * plugins
    * dart plugins
    * native plugins
  * ffi packages

Reply:
> I guess we now have two types of plugins; 1) using platform channels, and 2) using FFI.

It's even more complicated than that, because we have things like path_provider_linux which implement a plugin, but are purely Dart (using things like environment variables and config files). By itself it's just a regular package, but it's part of a plugin.

Reply:
I've adopted

* dart plugins
* native plugins
   * method channel
   * ffi
   
as nomenclature now.

Let me know if you'd prefer something else.

The registry in native code consequently is only for method channel plugins.